src/utils/swapMinAmounts.ts

```diff
@@ -90,43 +109,70 @@ export async function swapMinAmounts(
-
-  // Get an array of all the swap plugin names
-  const swapPluginNamesArr = Object.keys(account.swapConfig)
-
-  // Create a variable to store the swap quotes with the plugin name as the key
-  const minAmountObj: { [key: string]: number } = {}
-
-  // Loop over each plugin to attempt to get a swap quote
-  for (const plugin of swapPluginNamesArr) {
-    try {
-      const disabledPluginMap = createDisabledSwapPluginMap(
-        plugin,
-        swapPluginNamesArr
-      )
-      const requestOptions = { disabled: disabledPluginMap }
-      minAmountObj[plugin] = await findMinSwapAmount(
-        initBinarySearchMin,
-        initBinarySearchMax,
-        fromCurrencyWallet,
-        toCurrencyWallet,
-        requestOptions
-      )
-    } catch (e) {
-      console.log(e)
+  try {
+    // Enable all swap plugins
+    for (const swapPluginConfig of Object.values(account.swapConfig)) {
+      await swapPluginConfig.changeEnabled(true)
     }
-  }
 
-  return minAmountObj
+    // Get an array of all the swap plugin names
+    const swapPluginNamesArr = Object.keys(account.swapConfig)
+
+    // Initial parameters for findMinSwapAmount
+    // NOTE: This is a placeholder that will be replaced in the future
+    const initBinarySearchMin = 0
+    const fromCurrencyWallet = currencyWallets[0]
+    const fromWalletCurrencyCode: string =
+      fromCurrencyWallet.currencyInfo.currencyCode
+    const response = await fetch(
+      baseUri + route + queryStr + fromWalletCurrencyCode
+    )
+    const { exchangeRate } = asRatesServerResponse(await response.json())
+    const initBinarySearchMax = Math.ceil(
+      usdAmount *
+        parseFloat(exchangeRate) *
+        parseInt(fromCurrencyWallet.currencyInfo.denominations[0].multiplier)
+    )
+
+    // Loop over the currency wallets for the different destination currencies
+    for (const toCurrencyWallet of currencyWallets) {
+      if (fromCurrencyWallet === toCurrencyWallet) continue
+
+      const toWalletCurrencyCode: string =
+        toCurrencyWallet.currencyInfo.currencyCode
+
+      const currencyPair: string =
+        fromWalletCurrencyCode + '_' + toWalletCurrencyCode
+
+      // Create a variable to store the swap quotes with the plugin name as the key
+      const singleCurrencyPairMinAmtObj: SingleCurrencyPairMinAmt = {}
+
+      // Loop over each plugin to attempt to get a swap quote
+      for (const plugin of swapPluginNamesArr) {
+        try {
+          const disabledPluginMap = createDisabledSwapPluginMap(
+            plugin,
+            swapPluginNamesArr
+          )
+          const requestOptions = { disabled: disabledPluginMap }
+          singleCurrencyPairMinAmtObj[plugin] = await findMinSwapAmount(
+            initBinarySearchMin,
+            initBinarySearchMax,
+            fromCurrencyWallet,
+            toCurrencyWallet,
+            requestOptions
+          )
+        } catch (e) {
+          console.log(e)
+        }
+      }
```

Comment:
Triple nested for loops aren't the most readable.  Maybe consider turning the last loop into a function.

Reply:
I actually agree that it's very unreadable.
But maybe instead of creating another function, you might think about implementing it a bit differently and not needing 3 nested loops.

Reply:
New PRs created to address these comments.